dangerzone/cli.py

```diff
@@ -31,51 +31,23 @@ def cli_main(
 
     display_banner()
 
-    # Validate filename
-    valid = True
     try:
-        with open(os.path.abspath(filename), "rb") as f:
-            pass
-    except:
-        valid = False
-
-    if not valid:
-        click.echo("Invalid filename")
+        document = DocumentHolder(os.path.abspath(filename))
+    except DocumentFilenameException as e:
+        click.echo(str(e))
         exit(1)
 
-    document = DocumentHolder(os.path.abspath(filename))
-
     # Validate safe PDF output filename
-    if output_filename:
-        valid = True
-        if not output_filename.endswith(".pdf"):
-            click.echo("Safe PDF filename must end in '.pdf'")
-            exit(1)
-
-        try:
-            with open(os.path.abspath(output_filename), "wb"):
-                pass
-        except:
-            valid = False
-
-        if not valid:
-            click.echo("Safe PDF filename is not writable")
-            exit(1)
-
-        document.output_filename = os.path.abspath(output_filename)
-
-    else:
-        document.output_filename = (
-            f"{os.path.splitext(document.input_filename)[0]}-safe.pdf"
-        )
-        try:
-            with open(document.output_filename, "wb"):
-                pass
-        except:
-            click.echo(
-                f"Output filename {document.output_filename} is not writable, use --output-filename"
+    try:
+        if output_filename:
+            document.output_filename = os.path.abspath(output_filename)
+        else:
+            document.output_filename = (
+                f"{os.path.splitext(document.input_filename)[0]}-safe.pdf"
             )
-            exit(1)
+    except DocumentFilenameException as e:
+        click.echo(str(e))
+        exit(1)
```

Comment:
Why do we need to catch `DocumentFilenameException` in various places in the code? Can we catch them in the main function instead?

Reply:
This is done in the child PR #216. [These lines](https://github.com/freedomofpress/dangerzone/pull/216/commits/3be0b852efeaf784eae826a0e17aa92ce6e1c1ad#diff-1818f923cb4b1458022430fcaea85123d0a5d8fd15b270705b5ba7ba085938deL33-L46) specifically.

I'd prefer avoiding further complicating that rebase.

Reply:
Got it, thanks.